snprc_ehr/src/org/labkey/snprc_ehr/services/SNPRC_EHRValidator.java

```diff
@@ -68,8 +71,22 @@ public static void validateNewAnimalData(Container c, User u, NewAnimalData newA
         if (newAnimalData.getColony() == null)
             throw new ValidationException("Colony is required");
         // animalAccount
-        if (newAnimalData.getAnimalAccount() == null)
-            throw new ValidationException("Animal Account is required");
+        try
+        {
+            TableInfo ti = schema.getTable("ValidAccounts", schema.getDefaultContainerFilter());
+            SimpleFilter filter = new SimpleFilter(FieldKey.fromParts("account"), newAnimalData.getAnimalAccount(), CompareType.EQUAL);
```

Comment:
You want to make sure newAnimalData.getAnimalAccount() isn't null before passing it to the filter.

There is a TODO in SimpleFilter.java that says // TODO: _paramVals, getter, and callers should all be @NotNull. Consider List<?> as well

So in the future it may throw an exception